app/controllers/countries_controller.rb

```diff
@@ -4,6 +4,8 @@ class CountriesController < ApplicationController
   def show
     @country = GeoEntity.find_by(name: country_name_from_param(params[:name]))
 
+    return redirect_to(action: 'unprocessable_entity', controller: 'errors') unless @country
```

Comment:
I would probably redirect to not_found here. What do you think?

Reply:
Agreed (think I missed this after testing)